src/hooks/routing/useRouterTrade.ts

```diff
@@ -41,50 +42,49 @@ export function useRouterTrade(
 
+  // PRICE fetching is informational and costly, so it is done less frequently.
+  const pollingInterval = routerPreference === RouterPreference.PRICE ? ms`2m` : ms`15s`
+
   // Get the cached state *immediately* to update the UI without sending a request - using useGetQuoteQueryState -
   // but debounce the actual request - using useLazyGetQuoteQuery - to avoid flooding the router / JSON-RPC endpoints.
-  const { isError, data, currentData, fulfilledTimeStamp } = useGetQuoteQueryState(queryArgs)
-  const isValidBlock = useIsValidBlock(Number(data?.blockNumber))
-  const isSyncing = currentData !== data || !isValidBlock
+  const { isError, isFetching, data, currentData, fulfilledTimeStamp } = useGetQuoteQueryState(queryArgs)
 
-  // PRICE fetching is informational and costly, so it's done less frequently.
-  const pollingInterval = routerPreference === RouterPreference.PRICE ? ms`2m` : ms`15s`
+  // An already-fetched value should be refetched if it is older than the pollingInterval.
+  // Without explicit refetch, it would not be refetched until another pollingInterval has elapsed.
   const [trigger] = useLazyGetQuoteQuery({ pollingInterval })
   const request = useCallback(() => {
     const { refetch } = trigger(queryArgs, /*preferCacheValue=*/ true)
-    // An already-fetched value should be refetched if it is older than the pollingInterval.
-    // Without explicit refetch, it would not be refetched until another pollingInterval has elapsed.
-    if (fulfilledTimeStamp && Date.now() - fulfilledTimeStamp > pollingInterval) {
+    if (!isFetching && fulfilledTimeStamp && Date.now() - fulfilledTimeStamp > pollingInterval) {
       refetch()
     }
-  }, [fulfilledTimeStamp, pollingInterval, queryArgs, trigger])
+  }, [fulfilledTimeStamp, isFetching, pollingInterval, queryArgs, trigger])
   useTimeout(request, 200)
 
-  const [currencyIn, currencyOut] = isExactInput(tradeType)
-    ? [amountSpecified?.currency, otherCurrency]
-    : [otherCurrency, amountSpecified?.currency]
+  const result = useMemo(() => (typeof data === 'object' ? data : undefined), [data])
   const trade = useMemo(() => {
-    const routes = computeRoutes(currencyIn, currencyOut, tradeType, data)
+    const [currencyIn, currencyOut] = isExactInput(tradeType)
+      ? [amountSpecified?.currency, otherCurrency]
+      : [otherCurrency, amountSpecified?.currency]
+    const routes = computeRoutes(currencyIn, currencyOut, tradeType, result)
     if (!routes || routes.length === 0) return
     try {
       return transformRoutesToTrade(routes, tradeType)
     } catch (e: unknown) {
       console.debug('transformRoutesToTrade failed: ', e)
       return
     }
-  }, [currencyIn, currencyOut, data, tradeType])
-  const gasUseEstimateUSD = useStablecoinAmountFromFiatValue(data?.gasUseEstimateUSD)
+  }, [amountSpecified?.currency, otherCurrency, result, tradeType])
+  const isValidBlock = useIsValidBlock(Number(result?.blockNumber))
```

Comment:
will this cast hurt memoizing? i think it'll do number compares by value instead of reference but just checking

Reply:
yes, primitive comparisons (number, string, boolean) should always be by value.